src/esm_runscripts/workflow.py

```diff
@@ -1,8 +1,588 @@
-import sys, copy, os
+import sys
+import copy
 import esm_parser
 
+# from pprint import pprint
+
+import pdb
+
+
+class Workflow:
+    """A workflow class."""
+
+    def __init__(self, phases, always_run_with=[], phases_to_submit_to_batch_system=[]):
+        """
+        Create a new workflow.
+
+        Parameters
+        ----------
+        phases : list
+            List of workflow phases names
+        always_run_with : list
+            List of phases that precedes each phase in phases
+
+        Returns
+        -------
+        none
+        """
+        self.user_phases = []
+        self.first_task_in_queue = None                     # needed
+        self.last_task_in_queue = None
+        self.next_run_triggered_by = None
+        # TODO: Call here the phase object ???
+        self.phases = phases
+        self.always_run_with = always_run_with
+        self.phases_to_submit_to_batch_system = phases_to_submit_to_batch_system
+
+    @property
+    def num_phases(self):
+        """
+        Return the number of phases in workflow.
+        """
+        return len(self.phases)
+
+    def get_workflow_phase_by_name(self, phase_name):
+        """
+        Returns phase of phase_name
+
+        Arguments
+        ---------
+            self : class Workflow
+            phase_name : str (name of the phase to be returned
+
+        Returns
+        -------
+            phase : class phase or user_phase
+        """
+        for phase in self.phases + self.user_phases:
+            if phase.name == phase_name:
+                return phase
+
+    def get_phases_attribs_list(self, phase_type, attrib):
+        """
+        Returns a certain attribute for all phases as a list.
+
+        Parameters
+        ----------
+            self: class Workflow
+            phase_type: str (default or user)
+            attrib: str
+
+        Returns
+        -------
+            phases_attribs : list
+        """
+        if phase_type == 'user':
+            phases_attribs = [getattr(phase, attrib) for phase in self.user_phases]
+        else:
+            phases_attribs = [getattr(phase, attrib) for phase in self.phases]
+
+        return phases_attribs
+
+    def init_default_workflow(self, config):
+        """
+        Add workflow for precompute, compute, and tidy phases
+        etc information already here!
+
+        Parameters
+        ----------
+            self : Workflow object
+            config : dict
+
+        Returns
+        -------
+            self : Workflow object
+        """
+
+        workflow_phases = self.phases
+
+        # Calculating the number of tasks for each component/model
+        # needed for phase compute
+        tasks = calc_number_of_tasks(config)
+        # Initiate/create default workflow phase objects
+        # and reset/append to Workflow.phases variable
+        self.phases = []
```

Comment:
```suggestion
```

I would remove this. What if I want to initialize it, do something with it and then it gets here removed, when I expected to be appended? The initialization of this already happened in `__init__`